packages/cli/src/lib/setup/setupSetup.ts

```
@@ -182,164 +233,172 @@ Please DO NOT copy files manually into ioBroker storage directories!`
+            } catch {
+                // ignore
+            }
+            if (!existingObj || existingObj._id.startsWith('_design/')) {
+                obj.from = `system.host.${tools.getHostName()}.cli`;
+                obj.ts = Date.now();
+                await this.objects.setObjectAsync(obj._id, obj);
+                console.log(`object ${obj._id} ${!existingObj ? 'created' : 'updated'}`);
+                setTimeout(this.dbSetup, 25, iopkg, ignoreExisting, callback);
+            } else {
+                !ignoreExisting && console.log(`object ${obj._id} yet exists`);
+                setTimeout(this.dbSetup, 25, iopkg, ignoreExisting, callback);
+            }
         } else {
-            await tools.createUuid(objects);
+            await tools.createUuid(this.objects);
+            let configObj: ConfigObject | null | undefined;
             // check if encrypt secret exists
-            objects.getObject('system.config', (err, obj) => {
-                let configFixed = false;
-                if (obj && obj.type !== 'config') {
-                    obj.type = 'config';
-                    obj.from = `system.host.${tools.getHostName()}.cli`;
-                    obj.ts = Date.now();
-                    configFixed = true;
-                }
-                if (obj && (!obj.native || !obj.native.secret)) {
-                    require('crypto').randomBytes(24, (ex, buf) => {
-                        obj.native = obj.native || {};
-                        obj.native.secret = buf.toString('hex');
-                        obj.from = `system.host.${tools.getHostName()}.cli`;
-                        obj.ts = Date.now();
-                        objects.setObject('system.config', obj, () => setupReady(obj, callback));
-                    });
+            try {
+                configObj = await this.objects.getObject('system.config');
+            } catch {
+                // assume non existing
+            }
```

One thing you could do here is
```ts
const configObj = await this.objects.getObject('system.config').catch(() => undefined);
```
It's a bit weird mixing await and catch, but I think its okay for this case.

---

We pretty often use the try empty catch syntax in the repo, so I tend to stick to it.

---

True.
Just wanted to raise this possibility, since the upside would be that you can define the variable as a const so TS knows it is defined in callbacks.